includes/preprocessor_base.inc

```diff
@@ -107,20 +107,61 @@ abstract class IslandoraBatchPreprocessor {
   /**
    * Get an identifier we can use for an object.
    *
+   * @param AbstractFedoraObject $object
+   *   The object to be later processed and added to Fedora.
+   *
    * @return string
    *   An id to use in the back-end (Fedora).
    */
-  protected function getId() {
-    // Not specifying a namespace should result in one from the default
-    // declared in the fedora.fcfg ("changeme" by default).
-    $namespace = (isset($this->parameters['namespace'])) ?
-      $this->parameters['namespace'] :
-      NULL;
+  protected function getId(AbstractFedoraObject $object) {
+    // Command line argument overrides Collection Policy.
+    if (isset($this->parameters['namespace'])) {
+      $namespace = $this->parameters['namespace'];
+    }
+    else {
+      $namespace = $this->getNamespace($object);
+    }
 
     // TODO: Implement some form of caching, so we can get multiple at a time.
     return $this->connection->repository->api->m->getNextPid($namespace);
   }
 
+
+  /**
+   * Try to get the namespace from the collection policy.
+   *
+   * @param AbstractFedoraObject $object
+   *   The object to be later processed and added to Fedora.
+   *
+   * @return string|NULL
+   *   The namespace with which the object should be ingested.
+   *   NULL if we use the Fedora configured default.
+   */
+  protected function getNamespace(AbstractFedoraObject $object) {
+    // Get namespace based on collection.
+    $collection_policies =& drupal_static(__METHOD__, array());
+    $parent_pid = $this->parameters['parent'];
```

Comment:
Really, should be fine if we just jump out before we get here if we don't have `parent`... Something like:
```php
if (!isset($this->parameters['parent'])) {
  // Could explicitly return NULL, if desired?
  return;
}
$parent_pid = $this->parameters['parent'];
```

If we /really/ wanted to, we could make the implementation here deal with only `$this->parameters['namespace']` as it does now, and have the implementation looking at the collection policy in the scan batch preprocessor (which would make a call to `parent::getNamespace($object)` and use it if not `NULL` kind of stuff...